packages/@webex/plugin-meetings/src/statsAnalyzer/mqaUtil.ts

```diff
@@ -145,148 +373,183 @@ export const getVideoReceiverMqa = ({videoReceiver, statsResults, lastMqaDataSen
     : 0;
-  videoReceiver.common.rtpBitrate = videoReceiver.streams[0].common.receivedBitrate;
 
-  // From tracks //TODO: calculate a proper one
   const totalFrameReceivedInaMin =
     statsResults[mediaType][sendrecvType].framesReceived - lastFramesReceived;
   const totalFrameDecodedInaMin =
     statsResults[mediaType][sendrecvType].framesDecoded - lastFramesDecoded;
 
-  videoReceiver.streams[0].common.receivedFrameRate = Math.round(
+  videoReceiverStream.common.receivedFrameRate = Math.round(
     totalFrameReceivedInaMin ? totalFrameReceivedInaMin / 60 : 0
   );
-  videoReceiver.streams[0].common.renderedFrameRate = Math.round(
+  videoReceiverStream.common.renderedFrameRate = Math.round(
     totalFrameDecodedInaMin ? totalFrameDecodedInaMin / 60 : 0
   );
 
-  videoReceiver.streams[0].common.framesDropped =
+  videoReceiverStream.common.framesDropped =
     statsResults[mediaType][sendrecvType].framesDropped - lastFramesDropped;
-  videoReceiver.streams[0].receivedHeight = statsResults[mediaType][sendrecvType].height || 0;
-  videoReceiver.streams[0].receivedWidth = statsResults[mediaType][sendrecvType].width || 0;
-  videoReceiver.streams[0].receivedFrameSize =
-    (videoReceiver.streams[0].receivedHeight * videoReceiver.streams[0].receivedWidth) / 256;
+  videoReceiverStream.receivedHeight = statsResults[mediaType][sendrecvType].height || 0;
+  videoReceiverStream.receivedWidth = statsResults[mediaType][sendrecvType].width || 0;
+  videoReceiverStream.receivedFrameSize =
+    (videoReceiverStream.receivedHeight * videoReceiverStream.receivedWidth) / 256;
 
-  videoReceiver.streams[0].receivedKeyFrames =
+  videoReceiverStream.receivedKeyFrames =
     statsResults[mediaType][sendrecvType].keyFramesDecoded - lastKeyFramesDecoded || 0;
-  videoReceiver.streams[0].requestedKeyFrames =
+  videoReceiverStream.requestedKeyFrames =
     statsResults[mediaType][sendrecvType].totalPliCount - lastPliCount || 0;
 };
 
-export const getVideoSenderMqa = ({videoSender, statsResults, lastMqaDataSent, mediaType}) => {
+export const getVideoSenderMqa = ({videoSender, statsResults, lastMqaDataSent, baseMediaType}) => {
   const sendrecvType = STATS.SEND_DIRECTION;
 
-  const lastPacketsSent = lastMqaDataSent[mediaType]?.[sendrecvType].totalPacketsSent || 0;
-  const lastPacketsLost =
-    lastMqaDataSent[mediaType]?.[sendrecvType].totalPacketsLostOnReceiver || 0;
-  const lastBytesSent = lastMqaDataSent[mediaType]?.[sendrecvType].totalBytesSent || 0;
-  const lastKeyFramesEncoded =
-    lastMqaDataSent[mediaType]?.[sendrecvType].totalKeyFramesEncoded || 0;
-  const lastFirCount = lastMqaDataSent[mediaType]?.[sendrecvType].totalFirCount || 0;
-  const lastFramesSent = lastMqaDataSent[mediaType]?.[sendrecvType].framesSent || 0;
-  const {csi} = statsResults[mediaType];
-  if (csi && !videoSender.streams[0].common.csi.includes(csi)) {
-    videoSender.streams[0].common.csi.push(csi);
-  }
+  const lastPacketsSent = getTotalValueFromBaseType(
```

Comment:
large blocks of repetition, any way of simplifying?

Reply:
Updated.